doc/ReleaseNotes.md

```diff
@@ -1,5 +1,14 @@
 ﻿### 3.5.4 / 4.0.4-beta1
 * ICodeInspector API is obsolete.
+* ViewBag data can now be used in overwritten SetModel calls
+  Viewbag is set like the model on template creation (not in the ExecuteContext)
+  This leads to some minor breaking changes
+  - Creating a ExecuteContext with a non-null viewbag throws a NotSupportedException (obsolete API)
+  - added a DynamicViewBag parameter to IInternalTemplateService.Resolve (you should not need to use that API)
+  - ITemplate.SetModel has changed to ITemplate.SetData (you are not affected if you use TemplateBase or TemplateBase<T> as base class, which is recommended)
+  
+While these changes are (strictly speaking) breaking they should hardly affect anyone, so this update will get only a minor version jump.
+
```

Comment:
Thanks for your answer.
I think than we should rather get rid of semver alltogether, because it makes more sense to make the Razor2-Razor4 split. Semver cannot give us an answer for our situation where we want to adapt Razor4 as fast as possible (because it is an major change), but still support the stable Razor2 and Razor3 releases, right?

Increasing major would lead to a 4.0.0 release which doesn't match the previous 4.0.0 beta releases (which are Razor4), that doesn't make a lot of sense either IMHO.

Using the API in a weird way should be an exception to the rule here (because I see the above more as a bugfix than a new feature).
But that's exactly the reason why I'm asking this...
Going to 3.6.0 (even if there is no new API) is really the only thing possible at this point. Do you see any alternative?

Hell some people are still using Razor2 because they are limited to net40 but at least we can handle this situation by bundling them in one package (different RazorEngine.dll for net40 and net45). But we cannot bundle the Razor4 build in the package as well.

This is all a big mess and any hint to improve the situation is highly appreciated :)